### ISSUE TITLE
Pre/Post maps break unused-value and misplaced-docstring lints

### ISSUE THREAD

**Issue by IGJoshua** [clj-kondo/clj-kondo#2304]
- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**version**
clj-kondo v2024.03.13

**platform**
Native clj-kondo on Manjaro linux.

**editor**
Using clojure-lsp via eglot in Emacs.

**problem**
1. When a pre/post map is included in a function definition, no subsequent expression in the function's scope will report unused-value.
2. When a pre/post map is included in a function definition, an unused string after it is not reported as misplaced-docstring.

**repro**
```clojure
(defn foo [bar]
  {}
  "hello"
  (+ 2 2)
  bar)
```

**config**
This problem occurs even without a config.edn, and my test project does not have one.

**expected behavior**
I expect unused values after a pre/post map to be reported as unused-value, and for an unused string after a pre/post map to be reported as misplaced-docstring.
